internal/issue/comment/comment.go

```diff
@@ -3,29 +3,97 @@ package comment
 import (
 	"context"
 	"errors"
+	"fmt"
+	"os"
 	"strconv"
+	"strings"
+
+	"prcommenter/internal/common"
 
 	"github.com/google/go-github/github"
 )
 
+type Commenter struct {
+	client    GitHubClient
+	messageId string
+}
+
 type GitHubClient interface {
 	CreateComment(ctx context.Context, owner string, repo string, number int, comment *github.IssueComment) (*github.IssueComment, *github.Response, error)
+	ListComments(ctx context.Context, owner string, repo string, number int, opts *github.IssueListCommentsOptions) ([]*github.IssueComment, *github.Response, error)
+	EditComment(ctx context.Context, owner, repo string, commentID int64, comment *github.IssueComment) (*github.IssueComment, *github.Response, error)
+}
+
+func NewCommenter(client GitHubClient) *Commenter {
+	// Create a unique "id" embedded in the comment that identifies the comment generated by this pipeline+job
+	// The generated "id" is unique to the specific pipeline+job, with the plugin message-id param allowing further uniqueness
+	messageId := fmt.Sprintf("%s:%s:pr-commenter-buildkite-plugin", os.Getenv("BUILDKITE_PIPELINE_SLUG"), os.Getenv("BUILDKITE_LABEL"))
+	uniqueId, found := os.LookupEnv(common.PluginPrefix + "MESSAGE_ID")
+	if found {
+		messageId = fmt.Sprintf("%s:%s", messageId, uniqueId)
+	}
+
+	return &Commenter{
+		client:    client,
+		messageId: messageId,
+	}
```

Comment:
```suggestion
func NewCommenter(client GitHubClient) *Commenter {
	// Create a unique "id" embedded in the comment that identifies the comment generated by this pipeline+job
	// The generated "id" is unique to the specific pipeline+job+step, with the plugin message-id param allowing further uniqueness
	stepId := os.Getenv("BUILDKITE_JOB_ID")
	if stepId == "" {
		stepId = os.Getenv("BUILDKITE_STEP_KEY")
	}
	messageId := fmt.Sprintf("%s:%s:%s:pr-commenter-buildkite-plugin", os.Getenv("BUILDKITE_PIPELINE_SLUG"), os.Getenv("BUILDKITE_LABEL"), stepId)
	uniqueId, found := os.LookupEnv(common.PluginPrefix + "MESSAGE_ID")
	if found {
		messageId = fmt.Sprintf("%s:%s", messageId, uniqueId)
	}

	return &Commenter{
		client:    client,
		messageId: messageId,
	}
```

I'd suggest something like this, so we can prevent collision - if two steps have the same label (or both have no label) this would result in them both having identical message IDs and could result in overwrites of sticky comments

Reply:
@JoeColeman95 On second look, what about just using [BUILDKITE_JOB_ID](https://buildkite.com/docs/pipelines/configure/environment-variables#BUILDKITE_JOB_ID) instead of `STEP_KEY` / `LABEL` here? If the `JOB_ID` value is consistent across pipeline builds, that'd be enough to make the comment unique to the Step / Job right?

Reply:
Yeah, that works @tcsullens

Reply:
>but I'd say that we should try and use the job-id at the top-level as that would be a unique UUID, and fall back using !found as you have with the other values here?

My hesitation with using `JOB_ID` is that it isn't consistent across pipeline builds/runs (at least for our dynamic pipelines), so while it is _unique_, in our case we'd get a different `JOB_ID` and thus `messageId` for the comment for each pipeline run, so wouldn't have a consistent `messageId` to lookup when trying to match a comment — we'd end up with something like `<!-- test-pipeline:test:uuid-1234:pr-commenter-buildkite-plugin -->` on build 1 and then `<!-- test-pipeline:test:uuid-4567:pr-commenter-buildkite-plugin -->` on build 2. 

Does that make sense, or am I missing something with the use of `BUILDKITE_JOB_ID`?